### ISSUE TITLE
SWIFT_VERSION added multiple times and cause failure in pod install

### ISSUE THREAD

**Issue by uchar** [AppLovin/AppLovin-MAX-Unity-Plugin#98]
In my project I'm using `NiceVibration `package, Now when I'm trying to export the project from unity, it always failed in `pod install` section and I have to go to Xcode and change SWIFT_VERSION from `5.1 5.0` to `5.1 `.

I investigate the issue and find out that NiceVibration package has this line of code : 

`pbxProject.SetBuildProperty(targetGUID, "SWIFT_VERSION", "5.1");`
And there is this line in your SDK and in this file `MaxPostProcessBuildiOS.cs` that cause the issue I have   : 
` project.AddBuildProperty(targetGuid, "SWIFT_VERSION", "5");`

**Comment by santoshbagadi** [comment]
Hi @uchar , we already have a fix for this in an unreleased version that will be release soon. Do you mind updating the `AddSwiftSupportIfNeeded` method in `MaxPostProcessBuildiOS.cs` file to the following to fix the issue in the mean time.

```
        private static void AddSwiftSupportIfNeeded(string buildPath, PBXProject project, string targetGuid)
        {
            var swiftFileRelativePath = "Classes/MAXSwiftSupport.swift";
            var swiftFilePath = Path.Combine(buildPath, swiftFileRelativePath);
            var maxMediationDirectory = PluginMediationDirectory;
            var hasSwiftLanguageNetworksInProject = SwiftLanguageNetworks.Any(network => Directory.Exists(Path.Combine(maxMediationDirectory, network)));

            // Remove Swift file if no need to support Swift
            if (!hasSwiftLanguageNetworksInProject)
            {
                if (File.Exists(swiftFilePath))
                {
                    MaxSdkLogger.UserDebug("Removing Swift file references.");

                    var fileGuid = project.FindFileGuidByRealPath(swiftFilePath, PBXSourceTree.Source);
                    if (!string.IsNullOrEmpty(fileGuid))
                    {
                        project.RemoveFile(fileGuid);
                        project.RemoveFileFromBuild(targetGuid, fileGuid);

                        FileUtil.DeleteFileOrDirectory(swiftFilePath);
                    }
                }

                return;
            }

            // Add Swift file
            CreateSwiftFile(swiftFilePath);
            var swiftFileGuid = project.AddFile(swiftFilePath, swiftFileRelativePath, PBXSourceTree.Source);
            project.AddFileToBuild(targetGuid, swiftFileGuid);

            // Add Swift version property if needed
            var swiftVersion = project.GetBuildPropertyForAnyConfig(targetGuid, "SWIFT_VERSION");
            if (string.IsNullOrEmpty(swiftVersion))
            {
                project.SetBuildProperty(targetGuid, "SWIFT_VERSION", "5.0");   
            }

            // Enable Swift modules
            project.AddBuildProperty(targetGuid, "CLANG_ENABLE_MODULES", "YES");
        }
```

**Comment by uchar** [comment]
Not work in my case , because it seems that NiceVibration also doesn't check if the version exist or not and their build script runs after yours

**Comment by santoshbagadi** [comment]
In that case, you would have to reach out to NiceVibration as well and see if they can add a similar fix to their plugin.

**Comment by santoshbagadi** [comment]
Hi @uchar , a fix for this issue has been released in our plugin version 5.0.0. Please let us know if you still run into this issue due to our plugin.
